### PR TITLE
Updated README to more clearly state that linguist runs in a Git repo

### PR DESCRIPTION
**Pull request by arkban**
Attempt to help others avoid the same mistake I made https://github.com/github/linguist/issues/4854 .

### CODE REVIEW

**Comment by arkban** [comment]
CC @lildude

File: README.md [hunk 1]
```diff
@@ -120,6 +122,19 @@ lib/linguist.rb
...
 ```
 
+#### Single file
+
+Alternatively you can find stats for a single file using the `github-linguist` executable.
+
+You can try running `github-linguist` on files in this repository itself:
+
+```console
+$ bundle exec bin/github-linguist grammars.yml
```
Comment (lildude) [inline README.md:132]: Whooops. I've just realised we detail global gem installation instructions and then suggest using bundler to exec which won't work.

Mind changing this to:

```suggestion
$ github-linguist grammars.yml
```

... and whilst you're at it, would you mind doing the same in the "Git Repository" section above? 🙇‍♂️

File: README.md [hunk 2]
```diff
@@ -161,6 +176,15 @@ Linguist detects the language of a file but the actual syntax-highlighting is po
 If you experience an issue with the syntax-highlighting on GitHub, **please report the issue to the upstream grammar repository, not here.**
 Grammars are updated every time we build the Linguist gem so upstream bug fixes are automatically incorporated as they are fixed.
 
+### I get an error when using Linguist on a directory that is not a Git repo
```
Comment (lildude) [inline README.md:179]:
```suggestion
### I get an error when using Linguist on a directory that is not a Git repository
```

File: README.md [hunk 3]
```diff
@@ -161,6 +176,15 @@ Linguist detects the language of a file but the actual syntax-highlighting is po
...
+
+When specifying a directory Linguist only works on Git repos -- at the root -- and only on committed changes. It's
+primary use is on GitHub.com which uses bare repositories and thus changes need to be committed as individual files don't show on the filesystem.
```
Comment (lildude) [inline README.md:182]: I think we can simplify this. How about a slight wording tweak:

```suggestion
Linguist only works on Git repositories and individual files. Its primary use is on GitHub.com which uses bare
repositories and thus changes need to be committed as individual files don't show on the filesystem.
```

File: README.md [hunk 4]
```diff
@@ -161,6 +176,15 @@ Linguist detects the language of a file but the actual syntax-highlighting is po
...
+
+As a work around you could initialise a temporary Git repo in your directory as demonstrated in this
```
Comment (lildude) [inline README.md:184]: We prefer not to abbreviate in documentation outside of code samples.

```suggestion
As a work around you could initialise a temporary Git repository in your directory as demonstrated in this
```

Comment (arkban) [reply README.md:184]: I can appreciate that.

**Comment by arkban** [comment]
Updated PR, I used your wording verbatim (since this is your repo not mine ; )

**Pull request merged**